ts/main.ts

```diff
@@ -0,0 +1,99 @@
+import { fetchTeams, Team } from './data';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const teamsTab = document.getElementById('teams-tab') as HTMLButtonElement;
+  const favoritesTab = document.getElementById(
+    'favorites-tab',
+  ) as HTMLButtonElement;
+  const teamsView = document.getElementById('teams-view') as HTMLElement;
+  const detailsView = document.getElementById('details-view') as HTMLElement;
+  const favoritesView = document.getElementById(
+    'favorites-view',
+  ) as HTMLElement;
+  const teamsContainer = document.getElementById(
+    'teams-container',
+  ) as HTMLElement;
+  const clubDetails = document.getElementById('club-details') as HTMLElement;
+  const backButton = document.getElementById(
+    'back-button',
+  ) as HTMLButtonElement;
+
+  const favorites: Team[] = [];
+
+  teamsTab.addEventListener('click', () => {
+    teamsView.classList.add('active');
+    detailsView.classList.remove('active');
+    favoritesView.classList.remove('active');
+  });
+
+  favoritesTab.addEventListener('click', () => {
+    teamsView.classList.remove('active');
+    detailsView.classList.remove('active');
+    favoritesView.classList.add('active');
+    renderFavorites();
+  });
```

Comment:
You could make a reuseable `viewSwap` function to do this rather than having multiple event listeners